block-producer/src/batch_builder/batch.rs

```diff
@@ -62,23 +64,52 @@ impl TransactionBatch {
         let produced_nullifiers =
             txs.iter().flat_map(|tx| tx.input_notes().iter()).cloned().collect();
 
-        let (created_notes, created_notes_smt) = {
-            let created_notes: Vec<NoteEnvelope> =
+        let (created_notes, created_notes_smt, created_note_envelopes) = {
+            let created_note_envelopes: Vec<NoteEnvelope> =
                 txs.iter().flat_map(|tx| tx.output_notes().iter()).cloned().collect();
 
+            let created_notes: Vec<NoteCreated> = txs
+                .iter()
+                .flat_map(|tx| {
+                    tx.output_notes().iter().map(|note| {
+                        if let Some(note_with_details) = tx.get_output_note_details(&note.note_id())
+                        {
+                            NoteCreated {
+                                batch_index: 0,
+                                note_index: 0,
+                                note_id: Some(note.note_id().into()),
+                                sender: Some(note.metadata().sender().into()),
+                                tag: note.metadata().tag().into(),
+                                details: Some(note_with_details.to_bytes()),
+                            }
+                        } else {
+                            NoteCreated {
+                                batch_index: 0,
+                                note_index: 0,
+                                note_id: Some(note.note_id().into()),
+                                sender: Some(note.metadata().sender().into()),
+                                tag: note.metadata().tag().into(),
+                                details: None,
+                            }
+                        }
+                    })
+                })
+                .collect();
+
```

Comment:
This will likely have to change after https://github.com/0xPolygonMiden/miden-base/pull/572. Maybe it's best to point at that branch for now to not have to redo this soonish.

Reply:
Thanks for the heads up, fixed the implementation.